roles/delete_nodepools/tasks/main.yml

```diff
@@ -22,7 +28,7 @@
     username: "{{ ionos_username }}"
     password: "{{ ionos_password }}"
     state: absent
-  loop: "{{ terraform_run.outputs.nodepool_zone1_id.value + terraform_run.outputs.nodepool_zone2_id.value }}"
+  loop: "{{ nodepools_to_delete }}"
   loop_control:
     loop_var: pool_id
-  when: not ansible_check_mode and terraform_run.outputs.keys() | difference(['cluster_id', 'nodepool_zone1_id', 'nodepool_zone2_id'])
```

Comment:
When gets evaluated inside the loop so this doesn't prevent errors when the loop variable is undefined